### ISSUE TITLE
Python version in conda yaml specs does not override currently running python version

### ISSUE THREAD

**Issue by tylerpotts** [Netflix/metaflow-nflx-extensions#19]
When using the below conda yaml:

example_env.yml
```
name: example
channels:
  - conda-forge
dependencies:
  - python==3.11.6
```

If my current environment is running a different version of python (EG 3.12) then the resolved environment will attempt to solve with both versions:

```
$ metaflow environment resolve -f example_env.yml 

...

metaflow_extensions.netflix_ext.plugins.conda.utils.CondaException: Conda command '['/opt/conda/condabin/mamba', 'create', '--prefix', '/tmp/tmpqulod7yx/prefix', '--dry-run', 'requests==>=2.21.0', 'python==3.11.6', 'boto3==>=1.14.0', 'python==3.12.0']' returned error (1); see pretty-printed error above
```

Based on the current implementation the solution is to specify the python version in the resolve command like so: 

```
$ metaflow environment resolve -f example_env.yml --python "3.11.6"
```

This was somewhat non-intuitive to me, as I don't expect my current running version of python to impact the version of python that is used to resolve a conda environment. I would also expect that my explicitly pinned python version in the yaml spec would override any default python versions

**Comment by romain-intel** [comment]
Hey. That's a good find. Yes, I hadn't considered having people specify the python dependency in the yml file. I can easily change that. Do you think the following makes sense:
Python version is (in order of priority):
- the one specified in the yml file or requirements.txt
- the one specified on the command line
- the one the user is running in.

If there is one in the yml/requirements.txt file and it is also specified on the command line, I would raise an error to avoid having some weird surprise.

Would this be more intuitive?

**Comment by romain-intel** [comment]
Fixed in #21
